strapi/strapi
# chore(tests): backport e2e config

### What does it do?

- Backport work done on v5/history to improve testing with env variables and future flags
- A `.env` file can be added to the root of the `e2e` and the tests will run with these environment values
- A `config/features.js` can be added to the `e2e/app-template` and it will be applied to the generated apps
- Runs future features in CI

### Why is it needed?
- To improve DX
- To test future features in the CI

### How to test it?

- Create a `.env` at the root of the `e2e` folder
```
STRAPI_FEATURES_FUTURE_RELEASES_SCHEDULING=true
STRAPI_LICENSE=your-license
```
- Add a `config/features.js` file to `e2e/app-template`
```
module.exports = ({ env }) => ({
  future: {
    contentReleasesScheduling: env.bool('STRAPI_FEATURES_FUTURE_RELEASES_SCHEDULING', false),
  },
});
```
- Run 
```
yarn test:e2e --setup --domains=content-releases --concurrency=1 
```
- The tests should pass for releases (EE feature using a license) and scheduling (future feature)
- The tests should also pass in the CI


p.s.

My bad should have done this first and then merged into v5/main, but oh well. 

### Related issue(s)/PR(s)

- https://github.com/strapi/strapi/pull/19525
- https://github.com/strapi/strapi/pull/19407#issuecomment-1952416966
